object_database/PyDatabaseObjectType.cpp

```diff
@@ -1065,10 +1066,48 @@ PyObject* PyDatabaseObjectType::pyLookupAny(PyObject *databaseType, PyObject* ar
     });
 }
 
+PyObject* PyDatabaseObjectType::pyLookupUnique(PyObject *databaseType, PyObject* args, PyObject* kwargs) {
+    return translateExceptionToPyObject([&] {
+        if (PyTuple_Size(args)) {
+            throw std::runtime_error("lookupUnique does not accept positional arguments");
+        }
+
+        PyDatabaseObjectType* obType = PyDatabaseObjectType::check(databaseType);
+        if (!obType) {
+            throw std::runtime_error("Expected first argument to be a database type.");
+        }
+
+        View* view = View::currentView();
+        if (!view) {
+            throw std::runtime_error(
+                "Can't lookup instances of " + obType->m_schema_and_typename + " outside of a view."
+            );
+        }
+
+        std::pair<field_id, index_value> lookup = obType->parseIndexLookupKwarg(view, kwargs);
+
+        object_id oid = view->indexLookupFirst(lookup.first, lookup.second);
+
+        if (oid == NO_OBJECT) {
+            return incref(Py_None);
+        }
+        PyObject* result = obType->fromIntegerIdentity(oid);
+
+        // check uniqueness
+        oid = view->indexLookupNext(lookup.first, lookup.second, oid);
+        if (oid != NO_OBJECT) {
+            throw std::runtime_error(
```

Comment:
I think you may need to decref result here.

Reply:
Good catch. Added the decref

Reply:
Travis is happy. We're good to merge if you don't have additional comments.